app/controllers/users_controller.rb

```diff
@@ -30,6 +30,7 @@ def index
     else
       @users = User.joins(:role).where.not(id: User.first.id) #Remove first user
     end
+    @users = User.search(@users, params[:search]) if params[:search]
```

Comment:
What I meant was it would be better to include badge and role filtering in the `.search` method of the `User` model, instead of doing some of the filtering (badge/role) in the controller and then some of it (name) in the model. So just expand `.search` to accept badge and role parameters and use that for filtering, as well, allowing you to remove the code from the controller.

The way you implemented the `.search` method (requiring a list of users to filter more) it either shouldn't go in the model or should be implemented differently (without requiring the list). If you do all the filtering there, you won't need the list and it's fine to put it in the model code.